internal/api/gsi.go

```diff
@@ -0,0 +1,48 @@
+package api
+
+// IndexDDLRequest is a request to run an index DDL statement.
+type IndexDDLRequest struct {
+	// Definition The index DDL statement.  This can be a CREATE/DROP/ALTER/BUILD statement.
+	// Multiple delimited queries are not allowed.
+	Definition string `json:"definition"`
+}
+
+// QueryError is the error message returned by query service.
+type QueryError struct {
+	// Msg The error message.
+	Msg string `json:"msg"`
+}
+
+// IndexDDLResponse has an array of errors returned by query service.
+type IndexDDLResponse struct {
+	Errors []QueryError `json:"errors,omitempty"`
+}
+
+// IndexDefinitionResponse represents a single index definition.
+type IndexDefinitionResponse struct {
+	Bucket       string   `json:"bucket"`
+	Scope        string   `json:"scope"`
+	Collection   string   `json:"collection"`
+	IsPrimary    bool     `json:"is_primary"`
+	IndexName    string   `json:"indexName"`
+	SecExprs     []string `json:"secExprs"`
```

Comment:
Can you please add a GoDoc comment explaining what secExprs contains?

Reply:
Addressed in 5c6efe534d7761f3957ee6e5d1301148aba9aa2c